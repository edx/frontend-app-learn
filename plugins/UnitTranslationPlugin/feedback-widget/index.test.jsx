import { shallow } from '@edx/react-unit-test-utils';

import FeedbackWidget from './index';

jest.mock('@edx/paragon', () => jest.requireActual('@edx/react-unit-test-utils').mockComponents({
  ActionRow: {
    Spacer: 'Spacer',
  },
  IconButton: 'IconButton',
  Icon: 'Icon',
}));
jest.mock('@edx/paragon/icons', () => ({
  Close: 'Close',
  ThumbUpOutline: 'ThumbUpOutline',
  ThumbDownOffAlt: 'ThumbDownOffAlt',
}));
jest.mock('./useFeedbackWidget', () => () => ({
  closeFeedbackWidget: jest.fn().mockName('closeFeedbackWidget'),
  openFeedbackWidget: jest.fn().mockName('openFeedbackWidget'),
  sendFeedback: jest.fn().mockName('sendFeedback'),
  showFeedbackWidget: true,
  showGratitudeText: false,
}));
jest.mock('@edx/frontend-platform/i18n', () => {
  const i18n = jest.requireActual('@edx/frontend-platform/i18n');
  const { formatMessage } = jest.requireActual('@edx/react-unit-test-utils');
  // this provide consistent for the test on different platform/timezone
  const formatDate = jest.fn(date => new Date(date).toISOString()).mockName('useIntl.formatDate');
  return {
    ...i18n,
    useIntl: jest.fn(() => ({
      formatMessage,
      formatDate,
    })),
    defineMessages: m => m,
    FormattedMessage: () => 'FormattedMessage',
  };
});

describe('<FeedbackWidget />', () => {
  const props = {
    courseId: 'course-v1:edX+DemoX+Demo_Course',
    languageCode: 'es',
    unitId: 'block-v1:edX+DemoX+Demo_Course+type@vertical+block@37b72b3915204b70acb00c55b604b563',
    userId: '123',
  };
  it('renders', () => {
    const wrapper = shallow(<FeedbackWidget {...props} />);
    expect(wrapper.snapshot).toMatchSnapshot();
  });
});
