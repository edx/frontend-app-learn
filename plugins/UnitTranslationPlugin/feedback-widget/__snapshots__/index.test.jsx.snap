// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedbackWidget /> render feedback widget 1`] = `
<div
  className="d-none"
>
  <div
    className="sequence w-100"
  >
    <div
      className="ml-4 mr-2"
    >
      <ActionRow>
        Rate this page translation
        <Spacer />
        <div>
          <IconButton
            alt="positive-feedback"
            className="m-1"
            iconAs="Icon"
            id="positive-feedback-button"
            onClick={[MockFunction onThumbsUpClick]}
            src="ThumbUpOutline"
            variant="secondary"
          />
          <IconButton
            alt="negative-feedback"
            className="mr-2"
            iconAs="Icon"
            id="negative-feedback-button"
            onClick={[MockFunction onThumbsDownClick]}
            src="ThumbDownOffAlt"
            variant="secondary"
          />
        </div>
        <div
          className="mb-1 text-light action-row-divider"
        >
          |
        </div>
        <div>
          <IconButton
            alt="close-feedback"
            className="ml-1 mr-2 float-right"
            iconAs="Icon"
            id="close-feedback-button"
            onClick={[MockFunction closeFeedbackWidget]}
            src="Close"
            variant="secondary"
          />
        </div>
      </ActionRow>
    </div>
  </div>
</div>
`;

exports[`<FeedbackWidget /> render gratitude text 1`] = `
<div
  className="d-none"
>
  <div
    className="sequence w-100"
  >
    <div
      className="ml-4 mr-4"
    >
      <ActionRow
        className="m-2 justify-content-center"
      >
        Thank you! Your feedback matters.
      </ActionRow>
    </div>
  </div>
</div>
`;

exports[`<FeedbackWidget /> renders hidden by default 1`] = `
<div
  className="d-none"
>
  <div
    className="sequence w-100"
  >
    <div
      className="ml-4 mr-2"
    >
      <ActionRow>
        Rate this page translation
        <Spacer />
        <div>
          <IconButton
            alt="positive-feedback"
            className="m-1"
            iconAs="Icon"
            id="positive-feedback-button"
            onClick={[MockFunction onThumbsUpClick]}
            src="ThumbUpOutline"
            variant="secondary"
          />
          <IconButton
            alt="negative-feedback"
            className="mr-2"
            iconAs="Icon"
            id="negative-feedback-button"
            onClick={[MockFunction onThumbsDownClick]}
            src="ThumbDownOffAlt"
            variant="secondary"
          />
        </div>
        <div
          className="mb-1 text-light action-row-divider"
        >
          |
        </div>
        <div>
          <IconButton
            alt="close-feedback"
            className="ml-1 mr-2 float-right"
            iconAs="Icon"
            id="close-feedback-button"
            onClick={[MockFunction closeFeedbackWidget]}
            src="Close"
            variant="secondary"
          />
        </div>
      </ActionRow>
    </div>
    <div
      className="ml-4 mr-4"
    >
      <ActionRow
        className="m-2 justify-content-center"
      >
        Thank you! Your feedback matters.
      </ActionRow>
    </div>
  </div>
</div>
`;

exports[`<FeedbackWidget /> renders show when elemReady is true 1`] = `
<div
  className="sequence-container d-inline-flex flex-row"
>
  <div
    className="sequence w-100"
  >
    <div
      className="ml-4 mr-2"
    >
      <ActionRow>
        Rate this page translation
        <Spacer />
        <div>
          <IconButton
            alt="positive-feedback"
            className="m-1"
            iconAs="Icon"
            id="positive-feedback-button"
            onClick={[MockFunction onThumbsUpClick]}
            src="ThumbUpOutline"
            variant="secondary"
          />
          <IconButton
            alt="negative-feedback"
            className="mr-2"
            iconAs="Icon"
            id="negative-feedback-button"
            onClick={[MockFunction onThumbsDownClick]}
            src="ThumbDownOffAlt"
            variant="secondary"
          />
        </div>
        <div
          className="mb-1 text-light action-row-divider"
        >
          |
        </div>
        <div>
          <IconButton
            alt="close-feedback"
            className="ml-1 mr-2 float-right"
            iconAs="Icon"
            id="close-feedback-button"
            onClick={[MockFunction closeFeedbackWidget]}
            src="Close"
            variant="secondary"
          />
        </div>
      </ActionRow>
    </div>
    <div
      className="ml-4 mr-4"
    >
      <ActionRow
        className="m-2 justify-content-center"
      >
        Thank you! Your feedback matters.
      </ActionRow>
    </div>
  </div>
</div>
`;
